pkg/cla/quicl/endpoint.go

```diff
@@ -0,0 +1,412 @@
+// SPDX-FileCopyrightText: 2022 Markus Sommer
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+package quicl
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/dtn7/cboring"
+
+	"github.com/dtn7/dtn7-go/pkg/bpv7"
+	"github.com/dtn7/dtn7-go/pkg/cla"
+	"github.com/dtn7/dtn7-go/pkg/cla/quicl/internal"
+	"github.com/lucas-clemente/quic-go"
+	log "github.com/sirupsen/logrus"
+)
+
+// TODO: is this a reasonable value? I don't know...
```

Comment:
The default TCP connection timeout is 20s, which is really long, but 0.5s might be a bit short. Any testing we can do to figure out an appropriate value?